### DISCUSSION TITLE
User feedback : Alternate Identifier and Other cover image search.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#220] [General]
we have received the 400 documents of the semester and I test BiblioteQ on this arrival :-)

the addition of Alternate Identifier is very useful to get the covers on the BnF
result : on 18 ISBN starting with 979, 14 are OK

remarks for improvements : 
- don't allow isbn13 to isbn10 conversion via FROM ISBN-13 button if it starts with 979
- add search via amazon for 979...

Thanks again for the past and future developments

[User_Feeback_ISBN_979_and_Alternate_Identifier.sqlite.zip](https://github.com/textbrowser/biblioteq/files/9181260/User_Feeback_ISBN_979_and_Alternate_Identifier.sqlite.zip)

**Comment by textbrowser** [comment]
The algorithmic conversion from i-13 to i-10 is identical for 978 and 979.

https://www.isbn-international.org/content/changes-united-states-isbn-prefixes

BQ allows the conversion. Correctness is the task of the librarian.

**Reply by meteos77** [reply]
extract : [with 979 does not have an equivalent 10-digit ISBN](https://www.isbn.org/about_isbn_standard) and therefore BiblioteQ should rather indicate "this is not possible".

**Reply by meteos77** [reply]
in your link : However, a 13-digit ISBN starting 979 is not backwards compatible and does not have an equivalent 10-digit ISBN. Trying to convert a 979 ISBN into a **10 digit ISBN will bring serious errors including duplicated ISBNs** and incorrect supply of orders. All ISBNs must be recorded in all systems and communicated to partners ONLY in their complete 13-digit form.

**Reply by textbrowser** [reply]
**BQ allows the conversion. Correctness is the task of the librarian.**

**Comment by textbrowser** [comment]
979 searches in Amazon for images?

**Reply by meteos77** [reply]
I suppose that BiblioteQ searches on amazon by isbn-10 because if the field is empty its answer is 
"In order to download a cover image, Alternate Identifier or ISBN-10 must be provided."
so for isbn-13 starting with 979, the conversion to isbn-10 is not possible, so the search on amazon is not possible. hence my suggestion

**Comment by textbrowser** [comment]
You want an automatic conversion? Yuck. Not adding it for the purpose of an image query. It would have to be a silent conversion.

**Reply by meteos77** [reply]
the suggestion was rather to block the conversion of something impossible according to the standard
979xxxxxxxxxx -> not isbn-10 if you press the convert button.
(nothing automatic)

the ISBN-13 -> ISBN-10 conversion works well and that's the main thing.

I will try to do a SQL check.

**Reply by textbrowser** [reply]
It's not algorithmically impossible. It's impossible in your mind.

**Comment by textbrowser** [comment]
I'm beginning to re-evaluate the purpose of software. Software should be as open as possible. Life-support software has overrides. Life support? Mixing medications, overrides with toxic environments, automated objects, etc.

With BQ, I try to have it allow you to do just about anything. It's up to you to determine if you're doing it well.

**Reply by meteos77** [reply]
_With BQ, I try to have it allow you to do just about anything. It's up to you to determine if you're doing it well._
great thanks

**Reply by meteos77** [reply]
your sentence is too clever for my translator :-)

**Reply by textbrowser** [reply]
A tool is only as good as the person exercising it.

**Reply by meteos77** [reply]
I like to use BiblioteQ : I am very satisfied with the result

the result for our arrival on 418 books
- 15 books without cover (great result via Amazon for many + BnF via Alternate Identifier + 1 OpenLibrary)

- 2926 clicks to get the covers (I hope an improvement for 6 months on this subject :-)

I will be able to make my flyers to communicate the new arrivals at the library for 1 year

- by age thanks to the target_audience field
- by major themes (dewey index or location)

thanks to textbrowser and BiblioteQ !

[2022-07-Base-avec-couvertures.sqlite.zip](https://github.com/textbrowser/biblioteq/files/9182542/2022-07-Base-avec-couvertures.sqlite.zip)

**Reply by meteos77** [reply]
![Les-Contes](https://user-images.githubusercontent.com/56431039/180819745-fdab0ecd-e0b8-4e0b-a210-69067dc7d646.png)

**Reply by meteos77** [reply]
![LesDocumentaires-Géographique-Histoires](https://user-images.githubusercontent.com/56431039/180819834-86f894dd-c611-443a-b9a7-09d6122c74cf.png)

**Reply by textbrowser** [reply]
> I like to use BiblioteQ : I am very satisfied with the result
> 
> the result for our arrival on 418 books
> 
>     * 15 books without cover (great result via Amazon for many + BnF via Alternate Identifier + 1 OpenLibrary)
> 
>     * 2926 clicks to get the covers (I hope an improvement for 6 months on this subject :-)
> 
> 
> I will be able to make my flyers to communicate the new arrivals at the library for 1 year
> 
>     * by age thanks to the target_audience field
> 
>     * by major themes (dewey index or location)
> 
> 
> thanks to textbrowser and BiblioteQ !
> 
> [2022-07-Base-avec-couvertures.sqlite.zip](https://github.com/textbrowser/biblioteq/files/9182542/2022-07-Base-avec-couvertures.sqlite.zip)

Batch Activities may include (in the future) a feature of downloading cover images. The queries may include multiple sub-queries because the variety of identifiers.

**Reply by meteos77** [reply]
the batch activities will be a great evolution

**Reply by textbrowser** [reply]
Indeeeeeeeeed.

**Comment by textbrowser** [comment]
This approach is the most difficult. It's easy to restrict things. Democracy is complicated. It requires a wise environment. See?
